plugins/api/include/irods/plugins/api/switch_user_types.h

```diff
@@ -1,15 +1,82 @@
 #ifndef IRODS_SWITCH_USER_TYPES_H
 #define IRODS_SWITCH_USER_TYPES_H
 
+#include "irods/objInfo.h"
+
+/// The input data type used to invoke rc_switch_user.
+///
+/// \since 4.3.1
 typedef struct SwitchUserInput // NOLINT(modernize-use-using)
 {
-    // NOLINTBEGIN(modernize-avoid-c-arrays)
+    // NOLINTBEGIN(cppcoreguidelines-avoid-c-arrays, modernize-avoid-c-arrays)
+    // NOLINTBEGIN(cppcoreguidelines-avoid-magic-numbers, readability-magic-numbers)
+
+    /// The part of a fully-qualified iRODS username preceding the pound sign.
+    ///
+    /// The following requirements must be satisfied:
+    /// - The length must not exceed sizeof(UserInfo::userName) - 1 bytes
+    /// - It must be null-terminated
+    /// - It must be non-empty
+    /// - It must not include the pound sign
+    /// - It must not include the zone
+    ///
+    /// \since 4.3.1
     char username[64];
+
+    /// The part of a fully-qualified iRODS username following the pound sign.
+    ///
+    /// The following requirements must be satisfied:
+    /// - The length must not exceed sizeof(UserInfo::rodsZone) - 1 bytes
+    /// - It must be null-terminated
+    /// - It must be non-empty
+    /// - It must not include the pound sign
+    /// - It must not include the username
+    ///
+    /// \since 4.3.1
     char zone[64];
-    // NOLINTEND(modernize-avoid-c-arrays)
+
+    // NOLINTEND(cppcoreguidelines-avoid-magic-numbers, readability-magic-numbers)
+    // NOLINTEND(cppcoreguidelines-avoid-c-arrays, modernize-avoid-c-arrays)
+
+    /// TODO
+    ///
+    /// \since 4.3.1
+    struct KeyValPair options;
```

Comment:
`options` seems like a much better name than `condInput` to me. However, `condInput` is a common property name used across many APIs.

I think `options` does a better job at expressing the intent of the member variable.

Thoughts anyone?

Reply:
it's the name of the structure as well, right?   holding the bag of keywords?

Reply:
since this does the same as all the other things, `condInput` feels normal, even if we'd never choose that today, anew.

Reply:
Right. Usage would look like ...
```cpp
SwitchUserInput input{};
addKeyVal(&input.options, KEY_1, "");

// OR ...
irods::experimental::key_value_proxy kvp{input.options};
kvp[KEY_2] = "";
```